File: src/org/mtransit/parser/gtfs/data/GCalendarDatesExceptionType.java
package org.mtransit.parser.gtfs.data;

// https://developers.google.com/transit/gtfs/reference#calendar_dates_exception_type_field
public enum GCalendarDatesExceptionType {

	SERVICE_ADDED(1), SERVICE_REMOVED(2);

	public int id;

	GCalendarDatesExceptionType(int id) {
		this.id = id;
	}

	public static GCalendarDatesExceptionType parse(int id) {
		if (SERVICE_ADDED.id == id) {
			return SERVICE_ADDED;
		}
		if (SERVICE_REMOVED.id == id) {
			return SERVICE_REMOVED;
		}
		return SERVICE_ADDED; // default
	}

	public static GCalendarDatesExceptionType parse(String id) {
		if (id == null) { // that's OK
			return SERVICE_ADDED; // default
		}
		return parse(Integer.valueOf(id));
	}
	
	@Override
	public String toString() {
		return String.valueOf(id);
	}
}

File: src/main/java/org/mtransit/parser/gtfs/data/GCalendarDatesExceptionType.kt
package org.mtransit.parser.gtfs.data

import org.mtransit.commons.GTFSCommons

// https://gtfs.org/documentation/schedule/reference/#calendar_datestxt
enum class GCalendarDatesExceptionType(val id: Int) {

    SERVICE_DEFAULT(GTFSCommons.EXCEPTION_TYPE_DEFAULT), // from calendar (added by MT)
    SERVICE_ADDED(GTFSCommons.EXCEPTION_TYPE_ADDED),
    SERVICE_REMOVED(GTFSCommons.EXCEPTION_TYPE_REMOVED),
    ;

    companion object {

        fun parse(id: Int): GCalendarDatesExceptionType {
            return when {
                SERVICE_DEFAULT.id == id -> SERVICE_DEFAULT
                SERVICE_ADDED.id == id -> SERVICE_ADDED
                SERVICE_REMOVED.id == id -> SERVICE_REMOVED
                else -> SERVICE_ADDED // default
            }
        }

        @JvmStatic
        fun parse(id: String?): GCalendarDatesExceptionType {
            return if (id.isNullOrEmpty()) { // that's OK
                SERVICE_ADDED // default in GTFS spec
            } else parse(id.toInt())
        }
    }
}